### PR TITLE
test: Consolidate emsg tests

### PR DESCRIPTION
**Pull request by joeyparrish**
Some of the EMSG tests were redundant, and repeated for different timestamps.  Instead, run all EMSG tests with a non-zero timestamp offset, since 0 is just a degenerate case.

This also consolidates all of the test vectors so that the tests are easier to read.

### CODE REVIEW

**Comment by joeyparrish** [comment]
This is an indirect step toward restoring tests on Xbox, since I am digging into the effect these particular tests have on Xbox disconnect errors.

File: test/media/streaming_engine_unit.js [hunk 1]
```diff
@@ -2743,36 +2802,41 @@ describe('StreamingEngine', () => {
...
     it('raises an event for registered embedded emsg boxes', async () => {
-      segmentData[ContentType.VIDEO].segments[0] = emsgSegment;
-      videoStream.emsgSchemeIdUris = [emsgObj.schemeIdUri];
+      setSegment0(emsgSegmentV0);
+      videoStream.emsgSchemeIdUris = [emsgObjV0.schemeIdUri];
 
       // Here we go!
       streamingEngine.switchVariant(variant);
       streamingEngine.switchTextStream(textStream);
+      console.log('Skipping native screenshot tests on Safari < 16');
```
Comment (avelad) [inline test/media/streaming_engine_unit.js:2826]: this log has nothing to do with this test.

**Review by avelad — CHANGES_REQUESTED** [review]

File: test/media/streaming_engine_unit.js (hunk 1 above)
Comment (joeyparrish) [reply test/media/streaming_engine_unit.js:2826]: I must have accidentally pasted it here and overlooked it.  Thanks!

**Comment by joeyparrish** [comment]
I started this change ages ago, and just rebased today.  I think I made some mistakes in fixing merge conflicts.

**Review by littlespex — DISMISSED** [review]

**Review by theodab — DISMISSED** [review]

**Comment by avelad** [comment]
@joeyparrish, I fixed the PR to work properly, please review the changes. Thanks!

**Review by avelad — APPROVED** [review]

**Review by littlespex — APPROVED** [review]

**Comment by joeyparrish** [comment]
Thank you!  LGTM

**Pull request merged**